dev-module.nix

```diff
@@ -14,13 +20,39 @@
         ];
         shellHook = ''
           # Configure this repo to ignore certain revisions in git blame
-          git config blame.ignoreRevsFile .git-blame-ignore-revs
+          git config blame.ignoreRevsFile ${ignoreRevsFile}
           ${config.pre-commit.installationScript}
         '';
       };
+
+      treefmt = {
+        projectRootFile = "flake.nix";
+
+        programs = {
+          nixfmt.enable = true;
+          prettier.enable = true;
+        };
+
+        settings = {
+          on-unmatched = "fatal";
+          global.excludes = [
+            "*.gitignore"
+            "*.png"
+            "*.svg"
+            "*.toml"
+            "site/_redirects"
+            "site/src/highlight.js"
+            ignoreRevsFile
+          ];
+        };
+      };
+
       pre-commit = {
         settings = {
-          hooks.nixfmt-rfc-style.enable = true;
+          hooks.nix-fmt = {
+            enable = true;
+            entry = "nix fmt -- --fail-on-change";
+          };
```

Comment:
If we could enable `treefmt` instead, we get to use a cached build of it in the pre-commit config, so that committing doesn't require any evaluation. I'd like to keep committing quick.

Reply:
Good catch. I think I got it. I'll go ahead and do the same in other projects.